data/base.go

```diff
@@ -0,0 +1,286 @@
+	}
+
+	if validationErrors := validator.Struct(base); len(validationErrors) > 0 {
+		errs.AppendError(validationErrors.GetError(validationFailureReasons))
+	}
+
+	return base, errs
+}
+
+var validationFailureReasons = validate.ErrorReasons{
+	timeStrTag:        timeStrValidationMsg,
+	timeZoneOffsetTag: fmt.Sprintf("TimezoneOffset needs to be in minutes and greater than %d and less than %d", tzValidationLowerLimit, tzValidationUpperLimit),
+}
+
+func PayloadValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	//a place holder for more through validation
+	if _, ok := field.Interface().(interface{}); ok {
+		return true
+	}
+	return false
+}
+
+func AnnotationsValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	//a place holder for more through validation
+	if _, ok := field.Interface().([]interface{}); ok {
+		return true
+	}
+	return false
+}
+
+func TimezoneOffsetValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if offset, ok := field.Interface().(int); ok {
+		if offset >= tzValidationLowerLimit && offset <= tzValidationUpperLimit {
+			return true
+		}
+	}
+	return false
+}
+
+func TimeObjectValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if timeObject, ok := field.Interface().(time.Time); ok {
+		return isTimeObjectValid(timeObject)
+	}
+	return false
+}
+
+func isTimeObjectValid(timeObject time.Time) bool {
+	return !timeObject.IsZero() && timeObject.Before(time.Now())
+}
+
+func TimeStringValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if timeString, ok := field.Interface().(string); ok {
+		return isTimeStringValid(timeString)
+	}
+	return false
+}
+
+func isTimeStringValid(timeString string) bool {
+	var timeObject time.Time
+	timeObject, err := time.Parse(time.RFC3339, timeString)
+	if err != nil {
+		timeObject, err = time.Parse("2006-01-02T15:04:05", timeString)
+		if err != nil {
+			return false
+		}
+	}
+
+	return isTimeObjectValid(timeObject)
+}
+
+//Cast type for use in casting our incoming generic json data to the expected types that our data model uses
+type Cast struct {
+	err *Error
+}
+
+//NewCaster creates a Cast
+func NewCaster(err *Error) *Cast {
+	return &Cast{err: err}
+}
+
+//ToString will return the given data as a string or add an error to the cast obj
+func (cast *Cast) ToString(fieldName string, data interface{}) string {
+	if data == nil {
```

Comment:
Nil is not the same thing as passing an empty string.  We need to keep the two separate.